### DISCUSSION TITLE
Cannot implement protected interface member as protected?

### DISCUSSION THREAD

**Discussion by JesseRussell411** [dotnet/csharplang#7999] [Q&A]
I'm a little confused here. An interface can have a protected member, like this:
```
interface IFoo{
    protected int getInternalInt(); // <------------------ protected member

    public int useInternalInt(){
        return getInternalInt() + 1;
    }
}
```

But a class that implements the interface cannot make that member's implementation protected like this:
```
class Foo : IFoo {
    protected int getInternalInt() => 42;
}
```
![image](https://github.com/dotnet/csharplang/assets/47904783/8cbac2c6-8d72-4a18-9c9c-cbfa77ca9b6f)

It seems that the implementation has to be public. Then there's no error.
![image](https://github.com/dotnet/csharplang/assets/47904783/6c0bcf0b-600c-4d9b-bb6f-0d20f6433b42)

But why does it have to be public when the interface member is protected? The interface doesn't require the member to be public.

In fact, if you add a default implementation to the interface, then you *can* make the class's implementation protected. No error.
![image](https://github.com/dotnet/csharplang/assets/47904783/fb380e72-f7e3-4689-9e67-80f76c706d1e)

**Comment by KennethHoff** [comment]
In your default interface case; is that actually an implementation of the interface, or just a coincidental name overlap?

**Comment by zms9110750** [comment]
```csharp
interface IFoo
{
	protected int getInternalInt();

	public static void Hello(IFoo foo)
	{
		foo.getInternalInt();//How do we call it here? 
	}
}
class Foo : IFoo
{
	public int getInternalInt()//If this method is protected
	{
		throw new NotImplementedException();
	}
}
```
